python/pyarrow/_plasma.pyx

```diff
@@ -487,24 +491,36 @@ cdef class PlasmaClient:
 
         Returns
         -------
-        list of buffer objects or a buffer object
-            Python buffer or list of Python buffer for the data
-            associated with the object_ids and ObjectNotAvailable if the
-            object was not available.
+        list of tuples of metadata bytes and a buffer object
+            Python tuple or list of tuples of metadata bytes and a buffer
+            object associated with the object_ids and ObjectNotAvailable if
+            the object was not available.
         """
         cdef c_vector[CObjectBuffer] object_buffers
         if isinstance(object_ids, collections.Sequence):
             results = []
             self._get_object_buffers(object_ids, timeout_ms, &object_buffers)
             for i in range(object_buffers.size()):
                 if object_buffers[i].data.get() != nullptr:
-                    size = object_buffers[i].data.get().size()
-                    results.append(object_buffers[i].data.get().data()[:size])
+                    data = pyarrow_wrap_buffer(object_buffers[i].data)
                 else:
-                    results.append(None)
+                    data = ObjectNotAvailable
+                if object_buffers[i].data.get() != nullptr:
+                    metadata = object_buffers[i].metadata.get().data()[:]
+                else:
+                    metadata = None
+                results.append((metadata, data))
             return results
         else:
-            return self.get_buffer([object_ids], timeout_ms)[0]
+            return self.get_raw_buffer([object_ids], timeout_ms)[0]
+
+    def deserialize_buffer(self, buffer, serialization_context=None):
+        return pyarrow.deserialize(buffer, serialization_context)
+
+    def get_bytes_from_buffer(self, buffer):
+        buf = pyarrow_unwrap_buffer(buffer)
```

Comment:
Not a huge deal, but `buffer` is not the best variable name because it is a special keyword in python 2.